velox/exec/tests/QueryTraceTest.cpp

```diff
@@ -0,0 +1,222 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/init/Init.h>
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <memory>
+
+#include "velox/common/file/FileSystems.h"
+#include "velox/exec/PartitionFunction.h"
+#include "velox/exec/tests/utils/ArbitratorTestUtil.h"
+#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/exec/trace/QueryDataReader.h"
+#include "velox/exec/trace/QueryDataWriter.h"
+#include "velox/exec/trace/QueryMetadataReader.h"
+#include "velox/exec/trace/QueryMetadataWriter.h"
+#include "velox/serializers/PrestoSerializer.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+namespace facebook::velox::exec::test {
+class QueryTracerTest : public HiveConnectorTestBase {
+ public:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
+    HiveConnectorTestBase::SetUpTestCase();
+    if (!isRegisteredVectorSerde()) {
+      serializer::presto::PrestoVectorSerde::registerVectorSerde();
+    }
+    Type::registerSerDe();
+    common::Filter::registerSerDe();
+    connector::hive::HiveTableHandle::registerSerDe();
+    connector::hive::LocationHandle::registerSerDe();
+    connector::hive::HiveColumnHandle::registerSerDe();
+    connector::hive::HiveInsertTableHandle::registerSerDe();
+    core::PlanNode::registerSerDe();
+    core::ITypedExpr::registerSerDe();
+    registerPartitionFunctionSerDe();
+  }
+
+ protected:
+  void SetUp() override {
+    if (!isRegisteredVectorSerde()) {
+      serializer::presto::PrestoVectorSerde::registerVectorSerde();
+    }
+    filesystems::registerLocalFileSystem();
+    HiveConnectorTestBase::SetUp();
+  }
+
+  static VectorFuzzer::Options getFuzzerOptions() {
+    return VectorFuzzer::Options{
+        .vectorSize = 16,
+        .nullRatio = 0.2,
+        .stringLength = 1024,
+        .stringVariableLength = false,
+        .allowLazyVector = false,
+    };
+  }
+
+  QueryTracerTest() : vectorFuzzer_{getFuzzerOptions(), pool_.get()} {
+    filesystems::registerLocalFileSystem();
+  }
+
+  RowTypePtr generateTypes(size_t numColumns) {
+    std::vector<std::string> names;
+    names.reserve(numColumns);
+    std::vector<TypePtr> types;
+    types.reserve(numColumns);
+    for (auto i = 0; i < numColumns; ++i) {
+      names.push_back(fmt::format("c{}", i));
+      types.push_back(vectorFuzzer_.randType((2)));
+    }
+    return ROW(std::move(names), std::move(types));
+    ;
+  }
+
+  static bool checkPlan(
```

Comment:
Shall we just merge checkPlan into isSamePlan? thanks!

Reply:
Then let's remove checkPlan?

Reply:
removed